src/TensorOneD.cpp

```diff
@@ -0,0 +1,128 @@
+/*
+ * Copyright (c) 2015 - 2022, Intel Corporation
+ * SPDX-License-Identifier: BSD-3-Clause
+ */
+
+#include <math.h>
+#include "geopm/Exception.hpp"
+
+#include "TensorOneD.hpp"
+
+TensorOneD::TensorOneD()
+{
+}
+
+TensorOneD::TensorOneD(int dim)
+{
+    set_dim(dim);
+}
+
+TensorOneD::TensorOneD(const TensorOneD &other)
+{
+    set_dim(other.get_dim());
+    for (int idx = 0; idx < (int)m_vec.size(); idx++) {
+        m_vec[idx] = other[idx];
+    }
+}
+
+TensorOneD::TensorOneD(json11::Json input)
+{
+    set_dim(input.array_items().size());
+    for (int idx = 0; idx < (int)m_vec.size(); idx++) {
+        m_vec[idx] = input[idx].number_value();
+    }
+}
+
+int
+TensorOneD::get_dim() const
+{
+    return m_vec.size();
+}
+
+void
+TensorOneD::set_dim(int dim)
+{
+    m_vec.resize(dim);
+}
+
+TensorOneD
+TensorOneD::operator+(const TensorOneD& other)
+{
+    if (get_dim() != other.get_dim()) {
+        throw geopm::Exception("Adding vectors of mismatched dimensions.",
+                               GEOPM_ERROR_INVALID, __FILE__, __LINE__);
+    }
+
+    TensorOneD rval(m_vec.size());
+    for (int idx = 0; idx < (int)m_vec.size(); idx++) {
```

Comment:
Consider taking advantage of C++'s ``<algorithm>`` header. In this case, you could replace your ``for`` loop with:
``std::transform(m_vec.begin(), m_vec.end(), other.m_vec.begin(), rval.begin(), std::plus<float>())``

If you go that route, the following notes don't apply, but still worth mentioning for future reference:

- I would recommend avoiding the use of C-style casts. You could declare ``idx`` of your vector's ``size_type``. Alternatively, you could use ``size_t`` (technically, not necessarily the same type, but it should be a safe enough choice).
- Another approach could be to use iterators for the vectors, moving away from having to worry about indexing (considering no special indexing operations are being performed and you're iterating over both vectors in parallel).

Same comment applies to other ``for`` loops below.